encoding/encodingtest/series_iterator_split_into_blocks_test.go

```diff
@@ -0,0 +1,170 @@
+	start := now.Truncate(time.Hour).Add(2 * time.Minute)
+	end := start.Add(30 * time.Minute)
+
+	encoder := m3tsz.NewEncoder(start, checked.NewBytes(nil, nil), true, encoding.NewOptions())
+
+	i := 0
+	for at := start; at.Before(end); at = at.Add(time.Minute) {
+		datapoint := ts.Datapoint{Timestamp: at, Value: float64(i + 1)}
+		err := encoder.Encode(datapoint, xtime.Second, nil)
+		assert.NoError(t, err)
+		i++
+	}
+
+	iterAlloc := func(r io.Reader) encoding.ReaderIterator {
+		iter := m3tsz.NewDecoder(true, encoding.NewOptions())
+		return iter.Decode(r)
+	}
+
+	segment := encoder.Discard()
+
+	blockStart := start.Truncate(blockSize)
+	blockEnd := blockStart.Add(blockSize)
+
+	reader := xio.NewSegmentReader(segment)
+
+	multiReader := encoding.NewMultiReaderIterator(iterAlloc, nil)
+	multiReader.Reset([]xio.Reader{reader}, blockStart, blockEnd)
+
+	orig := encoding.NewSeriesIterator(ident.StringID("foo"), ident.StringID("namespace"),
+		start, end, []encoding.MultiReaderIterator{multiReader}, nil)
+
+	// Construct a per block view of the series
+	series := Series{
+		ID: orig.ID(),
+	}
+
+	// Collect all the replica per-block readers
+	for _, replica := range orig.Replicas() {
+		perBlockSliceReaders := replica.Readers()
+		next := true
+		for next {
+			// we are at a block
+			start := perBlockSliceReaders.CurrentStart()
+			end := perBlockSliceReaders.CurrentEnd()
+
+			var readers []xio.Reader
+			for i := 0; i < perBlockSliceReaders.CurrentLen(); i++ {
+				// reader to an unmerged (or already merged) block buffer
+				reader := perBlockSliceReaders.CurrentAt(i)
+
+				// import to clone the reader as we need its position reset before
+				// we use the contents of it again
+				clonedReader := reader.Clone()
+
+				readers = append(readers, clonedReader)
+			}
+
+			iter := encoding.NewMultiReaderIterator(iterAlloc, nil)
+			iter.Reset(readers, start, end)
+
+			inserted := false
+			for i := range series.Blocks {
+				if series.Blocks[i].Start.Equal(start) {
+					inserted = true
+					series.Blocks[i].Replicas = append(series.Blocks[i].Replicas, iter)
+					break
+				}
+			}
+			if !inserted {
+				series.Blocks = append(series.Blocks, SeriesBlock{
+					Start:    start,
+					End:      end,
+					Replicas: []encoding.MultiReaderIterator{iter},
+				})
+			}
+
+			next = perBlockSliceReaders.Next()
+		}
+	}
+
+	// Now per-block readers all collected, construct the per-block value
+	// iterator combining all readers from the different replica readers
+	for i, block := range series.Blocks {
+
+		filterValuesStart := orig.Start()
+		if block.Start.After(orig.Start()) {
+			filterValuesStart = block.Start
+		}
+
+		filterValuesEnd := orig.End()
+		if block.End.Before(filterValuesEnd) {
+			filterValuesEnd = block.End
+		}
+
+		valuesIter := encoding.NewSeriesIterator(orig.Namespace(), orig.ID(),
+			filterValuesStart, filterValuesEnd, block.Replicas, nil)
+
+		fmt.Printf("block at %v with %d replicas\n", block.Start, len(block.Replicas))
```

Comment:
Instead of printing output, can you change this to actually use assert/require to assert the values?